Async media: Design and copy updates

aforcier opened the pull request:
Addresses design and copy feedback from #6386. (Some things have been filed as separate issues, see [this to-do list](https://github.com/wordpress-mobile/WordPress-Android/issues/6386#issuecomment-318400411).)

* [x] Updated the snackbar text in the post list when uploads begin to say "Your post is uploading"
* [x] Updated the in-progress upload notification wording
* [x] Updated the wording on the successful upload notifications

Here's the current flow of uploading a post with media (asynchronously), showing the above changes:

![async-successful-media-post-upload-demo](https://user-images.githubusercontent.com/9613966/28719975-c774b3a8-7379-11e7-89b9-55e7d81738dc.gif)

([video link for the above](https://cloudup.com/cz6B-rstzRv))

Here are the rest of the success notifications:

![async-success-notifications](https://user-images.githubusercontent.com/9613966/28719685-cbeaaf06-7378-11e7-9097-7215d5493283.png)

* [x] Updated the wording on the post error and media error notifications:

![async-upload-error-notifications](https://user-images.githubusercontent.com/9613966/28720701-f050aa1e-737b-11e7-90cf-b5b4a1f37c8e.png)

* [x] Updated the wording of the post list error status for failed media (not the final fix, we should allow retries and give more specific error messages/suggestions when possible, tracked in https://github.com/wordpress-mobile/WordPress-Android/issues/6407)

![async-post-list-media-error](https://user-images.githubusercontent.com/9613966/28721008-d31228fa-737c-11e7-9de7-50bc35e76f79.png)

Here's a demo of the flow for a post upload with a media error:

![async-media-error-post-upload-demo](https://user-images.githubusercontent.com/9613966/28720467-44581d0a-737b-11e7-8679-6e670e7c77d4.gif)

([video link](https://cloudup.com/cRUmbKvmR0b))

cc @iamthomasbishop, @michelleweber, @mzorz

mzorz commented:
LGTM! Waiting for design and editorial approvals 👍

michelleweber commented:
On the "upload failed" messages/notis, can we add terminal punctuation -- a period a the very end?

For the failed media, can we say "the connection to our server was lost" rather than "connection error," to be more clear?

For the "unknown post" error/noti, the explanation is pretty cryptic. Is there any other way to describe it that's more layperson-friendly?

Otherwise, looking good!

mzorz commented:
@michelleweber regarding this one: 
>  can we say "the connection to our server was lost" 

it could also be that it is not our server we are trying to connect to, but a self-hosted site. How would you like to re-phrase it, taking this into account?

michelleweber commented:
Got it. In that case, just "the server" instead of "our server."

aforcier commented:
> are we dropping the user right on the specific error'd post in the posts list or just to the top of the list? I can't remember if we decided this was going to be address in a next iteration, or if this is already in place.

@iamthomasbishop I added the scrolling to post behavior in a separate PR: https://github.com/wordpress-mobile/WordPress-Android/pull/6424 (it has some more significant code changes than anything here, so I wanted to isolate it).

aforcier commented:
> For the "unknown post" error/noti, the explanation is pretty cryptic. Is there any other way to describe it that's more layperson-friendly?

@michelleweber We show that error when the user tried to edit a post, but it doesn't exist on the server. This probably means the post was deleted on the server. Should we say something like "Could not find the post on the server"?

Probably we should instead/also handle this case better, as there isn't really an option to fix this except to undelete the post on the server, or copy/paste the contents of the post to a new one (that would need to be done as part of the error handling improvements project).

michelleweber commented:
@aforcier, yeah -- the user still might not have a next step, but at least the error message is less technical and scary-sounding. It sounds more human, and therefore potentially fixable by someone.

aforcier commented:
Updated with all the copy changes:

* [x] Added terminal punctuation to error notification messages
* [x] Updated `"Connection error"` message to `"The connection to the server was lost"`
* [x] Updated `"Unknown post, local copy of this post may be out of sync with the server"` message to `"Could not find the post on the server"`

![async-updated-error-notifs](https://user-images.githubusercontent.com/9613966/28775477-3c920e7c-75c0-11e7-96b3-3e428937a4c9.png)

Thanks @michelleweber @iamthomasbishop! Looks like this is ready to go now @mzorz 🎉

mzorz commented:
:shipit:
